src/service/jobRoleService.ts

```diff
@@ -2,14 +2,14 @@ import axios from "axios";
 import type { JobRole } from "../model/jobRole";
 
 export const jobRoleService = {
-    async getJobRoles(): Promise<JobRole[]> {
+    async getJobRoles(token?: string): Promise<JobRole[]> {
+        if (!token) throw new Error("No token provided")
```

Comment:
What is shown to the user on this path? I think this error floats all the way to the top of the service